controllers/mattermost/mattermost/mattermost.go

```diff
@@ -22,42 +22,49 @@ import (
 	k8sClient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+type reconcileStatus struct {
+	ResourcesReady bool
+}
+
 func (r *MattermostReconciler) checkMattermost(
 	mattermost *mmv1beta.Mattermost,
 	dbInfo mattermostApp.DatabaseConfig,
 	fileStoreInfo *mattermostApp.FileStoreInfo,
 	status *mmv1beta.MattermostStatus,
-	reqLogger logr.Logger) error {
+	reqLogger logr.Logger) (reconcileStatus, error) {
 	reqLogger = reqLogger.WithValues("Reconcile", "mattermost")
+	recStatus := reconcileStatus{
+		ResourcesReady: true,
+	}
 
 	err := r.checkLicence(mattermost)
 	if err != nil {
-		return errors.Wrap(err, "failed to check mattermost license secret.")
+		return recStatus, errors.Wrap(err, "failed to check mattermost license secret.")
```

Comment:
nit: when we are returning the error I would suggest returning zero value of the struct, just to be transparent that it should not be used by the called in this case, otherwise it may be confusing if the `err != nil` but `ResourceReady == true`.
```suggestion
		return reconcileStatus{}, errors.Wrap(err, "failed to check mattermost license secret.")
```

Reply:
Yeah right thanks. I will return the `zero` value struct.